fuels-contract/src/contract.rs

```diff
@@ -159,7 +148,35 @@ impl Contract {
         if let Some(e) = encoded_args {
             script_data.extend(e)
         }
+        Ok((script, script_data))
+    }
 
+    /// Calls an already-deployed contract code.
+    /// Note that this is a "generic" call to a contract
+    /// and it doesn't, yet, call a specific ABI function in that contract.
+    /// We need a wallet to pay for the transaction fees (even though they are 0 right now)
+    #[allow(clippy::too_many_arguments)] // We need that many arguments for now
+    pub async fn call(
+        contract_id: ContractId,
```

Comment:
Might be worth factoring these args into a params struct with nice defaults and builder pattern.

Reply:
Definitely agree,tracking this in #190